src/app/log-in/log-in.component.ts

```diff
@@ -83,11 +80,31 @@ export class LogInComponent implements OnInit {
     });
   }
 
+  navigateToMetamaskSignup(): void {
+    this.router.navigate(['/', RouteNames.SIGN_UP_METAMASK], {
+      queryParamsHandling: 'merge',
+    });
+  }
+
   login(publicKey: string): void {
     this.identityService.login({
       users: this.accountService.getEncryptedUsers(),
       publicKeyAdded: publicKey,
       signedUp: false,
     });
   }
+
+  public getLoginIcon(loginMethod: LoginMethod): any {
+    return {
+      [LoginMethod.DESO]: 'assets/deso-logo.png',
+      [LoginMethod.GOOGLE]: 'assets/google_logo.svg',
+      [LoginMethod.METAMASK]: 'assets/metamask.png',
+    }[loginMethod];
+  }
 }
+
+export const getSpendingLimitsForMetamask = (): any => {
```

Comment:
Should the return type here be a TransactionSpendingLimit?

Reply:
Removed the any which will intrinsically the return type. Alternatively we could  add the deso-types lib and import the transaction spending limit object if we want to go the whole nine yards